lib/Runtime/Library/GlobalObject.cpp

```diff
@@ -2118,6 +2117,24 @@ namespace Js
         return TRUE;
     }
 
+    BOOL GlobalObject::DeleteProperty(JavascriptString *propertyNameString, PropertyOperationFlags flags)
+    {
+        PropertyRecord const *propertyRecord = nullptr;
```

Comment:
>PropertyRecord const *propertyRecord = nullptr; [](start = 8, length = 47)

This code is repeated at lot of places. Can it go in a static method that takes "this" as parameter and then you call "DeleteProperty" on this?